### PR TITLE
Make table and column names case insensitive

### PR DESCRIPTION
**Pull request by shrishrirang**

### CODE REVIEW

**Comment by shrishrirang** [comment]
@danderson00 - Closes https://github.com/Azure/azure-mobile-apps-js-client/issues/131

File: sdk/src/Platforms/cordova/MobileServiceSqliteStore.js [hunk 1]
```diff
@@ -63,21 +64,9 @@ var MobileServiceSqliteStore = function (dbName) { //TODO: allow null dbName
...
-        var columnDefinitions = tableDefinition.columnDefinitions;
-        for (var columnName in columnDefinitions) {
-            Validate.isString(columnDefinitions[columnName], 'columnType');
-            Validate.notNullOrEmpty(columnDefinitions[columnName], 'columnType');
-        }
-        
-        Validate.notNull(columnDefinitions[idPropertyName]);
+        storeHelper.validateTableDefinition(tableDefinition);
+
+        tableDefinition = JSON.parse( JSON.stringify( tableDefinition ) ); // clone the table definition as we will need it later
```
Comment (danderson00) [inline sdk/src/Platforms/cordova/MobileServiceSqliteStore.js]: nit: make spacing consistent (i.e. no spacing in function parameters)

File: sdk/src/Platforms/cordova/storeHelper.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+ */
+function getTableDefinition(tableDefinitions, tableName) {
+    Validate.isObject(tableDefinitions);
+    Validate.notNull(tableDefinitions);
+    Validate.isString(tableName);
+    Validate.notNullOrEmpty(tableName);
+
+    var tableDefinition = tableDefinitions[tableName.toLowerCase()];
+
+    if (tableDefinition) {
```
Comment (danderson00) [inline sdk/src/Platforms/cordova/storeHelper.js]: nit: just do return tableDefinitions[tableName.toLowerCase()]. The if statement is not required.

Comment (shrishrirang) [inline sdk/src/Platforms/cordova/storeHelper.js]: haha yes. WIll fix it. There was a throw 'table definition not found' after that which I removed before CR :)

**Comment by danderson00** [comment]
Looks good.

**Pull request merged**